source/Stargate-Model/WebOrigin.class.st

```diff
@@ -0,0 +1,67 @@
+Class {
+	#name : #WebOrigin,
+	#superclass : #Object,
+	#instVars : [
+		'scheme',
+		'host',
+		'port'
+	],
+	#category : #'Stargate-Model-CORS'
+}
+
+{ #category : #'instance creation' }
+WebOrigin class >> basedOn: aUrl [
+
+	( self hasValidOrigin: aUrl )
+		ifFalse: [ InstanceCreationFailed signal: ( '<1p> does not comply with a valid origin' expandMacrosWith: aUrl ) ].
+
+	^ self new initializeBasedOn: aUrl
+]
+
+{ #category : #preconditions }
+WebOrigin class >> hasValidOrigin: aZnUrl [
+	" According RFC6454, the Origin should have the format -- scheme '://' host [ ':' port ] --
```

Comment:
I would add a class comment similar to the one found in `WebLink` of Hyperspace, with a reference to the RFC instead of adding the comment on `#hasValidOrigin:`